modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -385,15 +385,25 @@ export class {{classname}} {
 
 {{/hasFormParams}}
 {{#useHttpClient}}
+    {{^isResponseFile}}
+        let responseType: 'text' | 'json' = 'json';
+        if(httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
+            responseType = 'text';
+        }
+
+    {{/isResponseFile}}
         return this.httpClient.{{httpMethod}}{{^isResponseFile}}<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>{{/isResponseFile}}(`${this.configuration.basePath}{{{path}}}`,{{#isBodyAllowed}}
```

Comment:
We'll need to strip the generics on 388/395 if we want to keep the type checking on responseType.

Reply:
Actually, there is no overload without the generics that accepts 'json' response type and including generics forces 'json'. So I'm not really seeing a way for this to work out without discarding type checks somewhere.

Reply:
Yeah, I guess there is no way to solve this, but your current code is fine, it enables using text responses. The returned observable is typed correctly anyway

Reply:
Yeah, I guess there is no way to solve this, but your current code is fine, it enables using text responses. The returned observable is typed correctly anyway